addons/controllers/tanzuclusterbootstrap_controller.go

```diff
@@ -348,11 +362,11 @@ func (r *TanzuClusterBootstrapReconciler) cloneSecretsAndProviders(cluster *clus
 	var createdSecrets []*corev1.Secret
 
 	packages := append([]*runtanzuv1alpha3.TanzuClusterBootstrapPackage{
-		bootstrap.Spec.CNI,
 		bootstrap.Spec.CPI,
 		bootstrap.Spec.CSI,
 		bootstrap.Spec.Kapp,
-	}, bootstrap.Spec.AdditionalPackages...)
+	}, bootstrap.Spec.CNIs...)
```

Comment:
cni selection should happen here

Reply:
I changed to do right before this function ```r.cloneSecretsAndProviders``` as ```createOrPatchTanzuClusterBootstrapFromTemplate``` seemed a better place to add and didn't need passing template object to ```cloneSecretsAndProviders```